static/js/avatar.js

```diff
@@ -66,16 +78,19 @@ export function build_user_avatar_widget(upload_function) {
         e.preventDefault();
         e.stopPropagation();
         function delete_user_avatar() {
+            display_avatar_delete_started();
             channel.del({
                 url: "/json/users/me/avatar",
                 success() {
-                    $("#user-avatar-upload-widget .image-delete-button").hide();
-                    $("#user-avatar-source").show();
+                    display_avatar_delete_complete();
                     // Need to clear input because of a small edge case
                     // where you try to upload the same image you just deleted.
                     get_file_input().val("");
                     // Rest of the work is done via the user_events -> avatar_url event we will get
                 },
+                error() {
+                    $("#user-avatar-upload-widget .image-delete-button").show();
```

Comment:
This needs to hide the spinner too, right?  You can simulate an error to confirm, but I believe you'll have a perpetual spinner with this implementation.

Reply:
Ah yes, missed it. This also needs to call `display_user_avatar_complete()`. I guess, I forgot to re-add that while testing. sorry.

Reply:
Changed!